lib/lists.rego

```diff
@@ -4,11 +4,7 @@ package lists
 append(l, x) = array.concat(l, [x])
 
 # Returns a new list with item x inserted at position p of list l
-insert(l, x, p) = r {
-	first := array.slice(l, 0, p)
-	last := array.slice(l, p, count(l))
-	r := array.concat(array.concat(first, [x]), last)
-}
+insert(l, x, p) = array.concat(array.concat(array.slice(l, 0, p), [x]), array.slice(l, p, count(l)))
```

Comment:
I was playing with this a bit and came up with another variant 😄 
```rego
insert_(l, x, p) = [at_index_or_list(i, p, x, l) | numbers.range(0, count(l)+2)[i]]

at_index_or_list(i, idx, _, list) = list[i] {
  i < idx
}
at_index_or_list(i, idx, _, list) = list[i-1] {
  i > idx
}
at_index_or_list(i, i, item, _) = item
```

Just sharing this, I don't think it's better. Merely less `array.*` 😀

Reply:
Fun stuff!